src/reactor.cc

```diff
@@ -0,0 +1,333 @@
+#include "reactor.h"
+
+#include "boost/shared_ptr.hpp"
+
```

Comment:
consider including some of these to reduce the need to fully specify the namespace of frequently used cyclus classes:
```
using cyclus::Material;
using cyclus::Composition;
using cyclus::toolkit::ResBuf;
using cyclus::toolkit::MatVec;
using cyclus::KeyError;
using cyclus::ValueError;
using cyclus::Request;
using cyclus::CompMap;
```